File: leetcode/cpp/1018_Binary_Prefix_Divisible_By_5.cpp
#include <vector>
using namespace std;
class Solution {
public:
	vector<bool> prefixesDivBy5(vector<int>& nums) {
		vector<bool> ret;
		int cur = 0;

		for (int i = 0; i < nums.size(); ++i) {
			cur = (cur * 2 + nums[i]) % 5;
			ret.push_back(cur == 0);
		}
		return ret;
	}
};

File: leetcode/cpp/1018_Binary_Prefix_Divisible_By_5.py
class Solution:
    def prefixesDivBy5(self, nums: List[int]) -> List[bool]:
        ret = []
        cur = 0

        for i in range(len(nums)):
            cur = (cur * 2 + nums[i]) % 5
            ret.append(cur == 0)

        return ret
